twenty-first/src/shared_math/polynomial.rs

```diff
@@ -116,75 +117,12 @@ impl<PFElem: FiniteField> PartialEq for Polynomial<PFElem> {
 
 impl<PFElem: FiniteField> Eq for Polynomial<PFElem> {}
 
-impl<PFElem: FiniteField> Polynomial<PFElem> {
-    pub fn new(coefficients: Vec<PFElem>) -> Self {
-        Self { coefficients }
-    }
-
-    pub fn new_const(element: PFElem) -> Self {
-        Self {
-            coefficients: vec![element],
-        }
-    }
-
-    // FIXME: Can be done with traits instead of explicitly mentioning B and X.
-    // Thor does not agree that this is a good path to venture down
-    pub fn lift_b_x(b_poly: &Polynomial<BFieldElement>) -> Polynomial<XFieldElement> {
-        let x_field_coefficients = b_poly.coefficients.iter().map(|b| b.lift()).collect();
-        Polynomial::new(x_field_coefficients)
-    }
-
-    pub fn normalize(&mut self) {
-        while !self.coefficients.is_empty() && self.coefficients.last().unwrap().is_zero() {
-            self.coefficients.pop();
-        }
-    }
-
-    pub fn zero() -> Self {
-        Self {
-            coefficients: vec![],
-        }
-    }
-
-    pub fn from_constant(constant: PFElem) -> Self {
-        Self {
-            coefficients: vec![constant],
-        }
-    }
-
-    pub fn is_zero(&self) -> bool {
-        self.coefficients.is_empty() || self.coefficients.iter().all(|x| x.is_zero())
-    }
-
-    pub fn is_one(&self) -> bool {
-        self.degree() == 0 && self.coefficients[0].is_one()
-    }
-
-    pub fn is_x(&self) -> bool {
-        self.degree() == 1 && self.coefficients[0].is_zero() && self.coefficients[1].is_one()
-    }
-
-    pub fn evaluate(&self, &x: &PFElem) -> PFElem {
-        let mut acc = PFElem::zero();
-        for &c in self.coefficients.iter().rev() {
-            acc = c + x * acc;
-        }
-
-        acc
-    }
-
-    pub fn leading_coefficient(&self) -> Option<PFElem> {
-        match self.degree() {
-            -1 => None,
-            n => Some(self.coefficients[n as usize]),
-        }
-    }
-
+impl<PFElem: FiniteField + MulAssign<BFieldElement>> Polynomial<PFElem> {
```

Comment:
Consider moving this to a `where` now that it's grown.

```suggestion
impl<PFElem> Polynomial<PFElem>
where
    PFElem: FiniteField + MulAssign<BFieldElement>
{
```

Reply:
Why not?